src/components/ConnectionHelp.js

```diff
@@ -0,0 +1,243 @@
+
+class ConnectionHelp extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      host: window.location.host,
+      open: false,
+    };
+  }
+
+  handleOpen = () => {
+    this.setState({
+      open: true,
+    });
+  }
+
+  handleClose = () => {
+    this.setState({
+      open: false,
+    });
+  }
+
+  render() {
+    const { open, host } = this.state;
+    let env;
+    switch (host) {
+      case 'beta.rovercode.com':
+        env = 'beta';
+        break;
+      case 'go.rovercode.com':
+        env = 'prod';
+        break;
+      case 'alpha.rovercode.com':
+      default:
+        env = 'alpha';
+    }
+
+    const PaddedBox = withStyles((theme) => ({
+      root: {
+        marginTop: theme.spacing(3),
+        marginBottom: theme.spacing(3),
+      },
+    }))(Box);
+
+    return (
+      <>
+        <IconButton color="inherit" onClick={this.handleOpen} aria-label="connection help">
+          <HelpOutline />
+        </IconButton>
+        <Dialog maxWidth="sm" open={open} onClose={this.handleClose}>
+          <DialogContent dividers>
+            <Typography variant="h4" gutterBottom>
+              <FormattedMessage
+                id="app.connection_help.welcome"
+                description="Introduces connection helper."
+                defaultMessage="Welcome! Let's get set up."
+              />
+            </Typography>
+            <Typography variant="h5" gutterBottom>
+              <FormattedMessage
+                id="app.connection_help.gigglebot_title"
+                description="Introduces the Gigglebot ordering section."
+                defaultMessage="1. Get your Rover"
+              />
+            </Typography>
+            <Typography gutterBottom>
+              <FormattedMessage
+                id="app.connection_help.gigglebot_ordering"
+                description="Tells the user they need a Gigglebot."
+                defaultMessage="The Rover is your robot. Rovercode works with the Gigglebot,
+                which is a robot chassis with a micro:bit computer."
+              />
+            </Typography>
+            <PaddedBox display="flex" justifyContent="center">
+              <img src={gigglebotImage} width="300px" alt="Gigglebot and micro:bit" />
+            </PaddedBox>
+            <PaddedBox display="flex" justifyContent="center">
+              <Button
+                color="primary"
+                variant="outlined"
+                href="https://gigglebot.io/shop/robots/gigglebot-starter-kit"
+                target="_blank"
+                rel="noopener"
+              >
+                <FormattedMessage
+                  id="app.connection_help.gigglebot_link"
+                  description="Links to Gigglebot kit"
+                  defaultMessage="If you don't already have a Gigglebot, click here to order one"
+                />
+              </Button>
+            </PaddedBox>
+            <Typography variant="h5" gutterBottom />
```

Comment:
Is this adding padding?

Reply:
Yup. I can't figure out how their Typography wants me to to spacing between sections. This is not great, but at least it's declarative. :shrug: